### DISCUSSION TITLE
Access to all calendar member in org with admin account

### DISCUSSION THREAD

**Discussion by conhv** [ecederstrand/exchangelib#1328] [Q&A] [answered]
Hi everyone
I want to collect all calendar (meeting) with all user in my org using admin account or account have permission to collect.
Do you know a method to get that?
Thank you.

**Comment by ecederstrand** [comment]
If you want to use exchangelib, then loop over your list of email addresses and create an `Account` object for each. Then connect to the calendar for that account. EWS does not support getting a list of all email addreses in an organisation, so you'll have to get that list via other means.

That said, maybe you're better off doing this by using some of the Exchange Server PowerShell admin commands?

**Reply by conhv** [reply]
Thank you for your answer
As i known with exchange have schedule assistant module to suggest a meeting with person be invited. I want to get the busy time from given list of person.
Do you known a method to get the busy/free time that i don’t have a permission get other user calendar details
Thank you again!!!!

**Reply by ecederstrand** [reply, marked as answer]
exchangelib has the `Aaccount.protocol.get_free_busy_info()` method you can use to get free/busy info. See https://ecederstrand.github.io/exchangelib/#non-account-services

**Reply by conhv** [reply]
Thank you
It very helpful for me.

**Reply by conhv** [reply]
Hi @ecederstrand 
I have one more question: my org have a hybrid exchange: cloud and on-premise do you have a method to get free and busy info at both side?

**Reply by ecederstrand** [reply]
That should just work. Are you experiencing problems?

**Reply by conhv** [reply]
When I connect account that account host on cloud it will redirect in to <user>@onmicrosof… and it need authourize for access
And i want to connect with office365, what server can i config, creds and permission i need?

**Reply by ecederstrand** [reply]
The documentation has a section on connecting to accounts on Office365: https://ecederstrand.github.io/exchangelib/#impersonation-oauth-on-office-365

**Reply by conhv** [reply]
Hi @ecederstrand  if i grant EWS.AccessAsUser.All, how permission effect? (Read mail message, calendars, ibox,…). I want only view free and busy info given user

**Reply by ecederstrand** [reply]
For questions regarding the permission sytem of Exchange, I would suggest asking in a forum specific to Exchange administration instead.
